curta/src/chip/hash/sha/sha256/mod.rs

```diff
@@ -27,15 +27,23 @@ pub type U32Value<T> = <U32Register as Register>::Value<T>;
 
 #[derive(Debug, Clone)]
 pub struct SHA256Gadget {
+    // each block 512 bit, split into 16 word, each stored in u32 register? 
+    // so actually this public_word should change for each block? or is it bc air constraint independent of the trace, the gadget only defined for one block's hash process, and periodically apply?
     /// The input chunks processed into 16-words of U32 values
     pub public_word: ArrayRegister<U32Register>,
+    // why there's 1024 round? is it bc for hash of dynamic length using same circuit?
```

Comment:
its due to the fact that we use a trace length of 2^16

Reply:
and each sha round requires 64 rows

Reply:
ok guess need to look into how lookup table works..